Source/AwsCommonRuntimeKit/mqtt/Mqtt5Client.swift

```diff
@@ -121,44 +121,154 @@ public typealias OnWebSocketHandshakeInterceptComplete = (HTTPRequestBase, Int32
+    /// The client will attempt to stay connected using the properties of the reconnect-related parameters
+    /// in the Mqtt5Client configuration on client creation.
+    ///
+    /// - Throws: CommonRuntimeError.crtError
+    public func start() throws {
+        try self.clientCore.start()
+    }
+
+    /// Notifies the Mqtt5Client that you want it to end connectivity to the configured endpoint, disconnecting any
+    /// existing connection and halting any reconnect attempts. No DISCONNECT packets will be sent.
+    ///
+    /// - Parameters:
+    ///     - disconnectPacket: (optional) Properties of a DISCONNECT packet to send as part of the shutdown
+    ///     process. When disconnectPacket is null, no DISCONNECT packets will be sent.
+    ///
+    /// - Throws: CommonRuntimeError.crtError
+    public func stop(disconnectPacket: DisconnectPacket? = nil) throws {
+        try self.clientCore.stop(disconnectPacket: disconnectPacket)
+    }
+
+    /// Tells the client to attempt to subscribe to one or more topic filters.
+    ///
+    /// - Parameters:
+    ///     - subscribePacket: SUBSCRIBE packet to send to the server
+    /// - Returns:
+    ///     - `SubackPacket`: return Suback packet if the subscription operation succeeded
+    ///
+    /// - Throws: CommonRuntimeError.crtError
+    public func subscribe(subscribePacket: SubscribePacket) async throws -> SubackPacket {
+        return try await clientCore.subscribe(subscribePacket: subscribePacket)
+    }
+
+    /// Tells the client to attempt to publish to topic filter.
+    ///
+    /// - Parameters:
+    ///     - publishPacket: PUBLISH packet to send to the server
+    /// - Returns:
+    ///     - For qos 0 packet: return `None` if publish succeeded
+    ///     - For qos 1 packet: return `PublishResult` packet if the publish succeeded
+    ///
+    /// - Throws: CommonRuntimeError.crtError
+    public func publish(publishPacket: PublishPacket) async throws -> PublishResult {
+        return try await clientCore.publish(publishPacket: publishPacket)
+    }
+
+    /// Tells the client to attempt to unsubscribe to one or more topic filters.
+    ///
+    /// - Parameters:
+    ///     - unsubscribePacket: UNSUBSCRIBE packet to send to the server
+    /// - Returns:
+    ///     - `UnsubackPacket`: return Unsuback packet if the unsubscribe operation succeeded
+    ///
+    /// - Throws: CommonRuntimeError.crtError
+    public func unsubscribe(unsubscribePacket: UnsubscribePacket) async throws -> UnsubackPacket {
+        return try await clientCore.unsubscribe(unsubscribePacket: unsubscribePacket)
+    }
+
+    /// Force the client to discard all operations and cleanup the client.
+    public func close() {
+        clientCore.close()
+    }
+
+    deinit {
+        clientCore.close()
+    }
+
+}
+
+// MARK: - Internal/Private
+
+/// Mqtt5 Client Core, internal class to handle Mqtt5 Client operations
+public class Mqtt5ClientCore {
```

Comment:
From what I understand the idea of Core class is to retain stuff that is needed while waiting for the callback from C. So that the actual class can go away, and when the callback happens, we don't crash with invalid references. Having a Core class which is just a thin wrapper over the actual class is not useful IMO and just makes the code harder to read.

Reply:
I make the native client and related members live in internal core instead of the actual class, so that we could avoid any accessing to the actual client class during callback or any native operations to avoid invalid references issue,